velox/type/DecimalUtil.h

```diff
@@ -193,6 +199,44 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  /// Rescale a double value to decimal value.
+  ///
+  /// Use `folly::tryTo` to convert a double value to int128_t or int64_t. It
+  /// returns an error when overflow occurs so that we could determine whether
+  /// an overflow occurs through checking the result.
+  ///
+  /// Normally, return the rescaled value. Otherwise, if the `toValue` overflows
+  /// the TOutput's limits or the `toValue` exceeds the precision's limits, it
+  /// will set an error message in `error` field and return an std::nullopt.
+  template <typename TOutput>
+  inline static folly::Expected<TOutput, ConversionErrorCode>
+  rescaleDouble(double value, int precision, int scale) {
+    if (!std::isfinite(value)) {
+      return folly::makeUnexpected<ConversionErrorCode>(
+          ConversionErrorCode::INFINITE_FLOATING_NUMBER);
```

Comment:
Is the error code suitable here? Because std::isfinite returns false for infinite and NaN.

Reply:
Yes, it's suitable here. The function`std::isfinite` returns false for `INFINITY` and `NAN`, that makes the judge condition be true, and both numbers are infinite floating numbers. So the error code`INFINITE_FLOATING_NUMBER` is suitable for these values.